sdk/tables/azure-data-tables/azure/data/tables/_table_client.py

```diff
@@ -318,24 +317,13 @@ def delete_entity(self, *args: Union[TableEntity, str], **kwargs) -> None:
             if row_key is None:
                 row_key = args[1]
 
-        match_condition = kwargs.pop("match_condition", None)
-        etag = kwargs.pop("etag", None)
-        if match_condition and entity and not etag:
-            try:
-                etag = entity.metadata.get("etag", None)
```

Comment:
We don't want to remove support for this - it would break existing behaviour and also is unrelated to the new code gen API

Reply:
Added them back